Framework/API/src/WorkspaceHistory.cpp

```diff
@@ -26,6 +31,25 @@ namespace API {
 namespace {
 /// static logger object
 Kernel::Logger g_log("WorkspaceHistory");
+struct AlgorithmHistorySearch {
```

Comment:
Why do these functors need to be `struct`? Wouldn't plain old function suffice?

Reply:
So the struct definitions are there so I can use them in an std::unordered_set, in comparison with a std::function call in some cases it can be much faster to define them as a struct and thus an operator() function because of inlining. In this case speed comes well before nice looking code as it can have detrimental effects on algorithms such as GroupWorkspaces or really anything that moves large amounts of history. I can only really cite https://stackoverflow.com/questions/25985248/speed-of-bound-lambda-via-stdfunction-vs-operator-of-functor-struct as reasoning here :)

Reply:
Fair enough. It's better to use functors then.